tasks/Task.cpp

```diff
@@ -152,7 +160,7 @@ void Task::updateHook()
 		    trajectories.erase(trajectories.begin());
 	    } else
 	    {
-		if(state() != REACHED_THE_END)
+		if(!(state() == REACHED_THE_END || state() == REACHED_THE_END_WRITING_STOPPED))
```

Comment:
You have to unfold the code ;-)

this lines are in the status switch and only executed in the state TrajectoryFollower::REACHED_TRAJECTORY_END

The REACHED_THE_END state is only set when the component is NOT in (state() == REACHED_THE_END || state() == REACHED_THE_END_WRITING_STOPPED), so no ping-pong here